dcap/types/src/certification_data.rs

```diff
@@ -132,13 +178,41 @@ mod test {
 
     extern crate alloc;
     use alloc::vec::Vec;
+    use yare::parameterized;
 
     #[test]
-    fn zero_certification_data() {
+    fn zero_type_is_error() {
         let bytes = [0u8; MIN_CERT_DATA_SIZE];
+        assert_eq!(
+            CertificationData::try_from(bytes.as_slice()),
+            Err(Quote3Error::CertificationDataType(0))
+        );
+    }
+
+    #[test]
+    fn type_eight_is_error() {
+        let mut bytes = [0u8; MIN_CERT_DATA_SIZE];
+        bytes[0] = 8;
+        assert_eq!(
+            CertificationData::try_from(bytes.as_slice()),
+            Err(Quote3Error::CertificationDataType(8))
+        );
+    }
+
+    #[parameterized(
+    one = {1, CertificationData::Ppid(&[])},
+    two = {2, CertificationData::PpidEncryptedRsa2048(&[])},
+    three = {3, CertificationData::PpidEncryptedRsa3072(&[])},
+    four = {4, CertificationData::Pck(&[])},
+    five = {5, CertificationData::PckCertChain(PckCertChain{data: &[]})},
+    six = {6, CertificationData::EcdsaSignatureAuxData(&[])},
+    seven = {7, CertificationData::PlatformManifest(&[])},
```

Comment:
Can you make these test names more descriptive? I can't easily tell what the numbers correspond to.

Reply:
updated to `byte_value_one` I'm up for suggestions on better names.  Trying to be concise and descriptive. Being descriptive it feels like "byte value of the certification data type on the wire"